File: app/src/main/java/androidstudio/class29/java/AdapterMusics.java
package androidstudio.class29.java;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.rafaelmfer.codeclasses.databinding.ItemClass29MusicsBinding;

import java.util.List;

import androidstudio.class29.java.pojo.Music;

public class AdapterMusics extends RecyclerView.Adapter<AdapterMusics.MusicsViewHolder> {

    private List<Music> musicList;

    AdapterMusics(List<Music> musicList) {
        this.musicList = musicList;
    }

    @NonNull
    @Override
    public MusicsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemClass29MusicsBinding itemClass29MusicsBinding = ItemClass29MusicsBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new MusicsViewHolder(itemClass29MusicsBinding);
    }

    @Override
    public int getItemCount() {
        return musicList.size();
    }

    @Override
    public void onBindViewHolder(@NonNull MusicsViewHolder holder, int position) {
        Music music = musicList.get(position);

        holder.musicTitleClass29.setText(music.getName());
        holder.bandNameClass29.setText(music.getBandArtist());
    }

    static class MusicsViewHolder extends RecyclerView.ViewHolder {

        TextView musicTitleClass29, bandNameClass29;

        MusicsViewHolder(@NonNull ItemClass29MusicsBinding itemClass29MusicsBinding) {
            super(itemClass29MusicsBinding.getRoot());
            musicTitleClass29 = itemClass29MusicsBinding.musicTitleClass29;
            bandNameClass29 = itemClass29MusicsBinding.bandNameClass29;
        }
    }
}

File: app/src/main/java/androidstudio/class29/kotlin/recycler/AdapterMusics.kt
package androidstudio.class29.kotlin.recycler

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidstudio.class29.java.pojo.Music
import androidx.recyclerview.widget.RecyclerView
import com.rafaelmfer.codeclasses.databinding.ItemClass29MusicsBinding

class AdapterMusics(private val musicList: List<Music>) : RecyclerView.Adapter<AdapterMusics.MusicsViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MusicsViewHolder =
        MusicsViewHolder(ItemClass29MusicsBinding.inflate(LayoutInflater.from(parent.context), parent, false))

    override fun getItemCount(): Int = musicList.size

    override fun onBindViewHolder(holder: MusicsViewHolder, position: Int) {
        musicList[position].run {
            holder.apply {
                musicTitleClass29.text = name
                bandNameClass29.text = bandArtist
            }
        }
    }

    class MusicsViewHolder(itemClass29MusicsBinding: ItemClass29MusicsBinding) : RecyclerView.ViewHolder(itemClass29MusicsBinding.root) {
        var musicTitleClass29: TextView = itemClass29MusicsBinding.musicTitleClass29
        var bandNameClass29: TextView = itemClass29MusicsBinding.bandNameClass29
    }
}